app/gui/src/project-view/components/visualizations/TableVisualization.vue

```diff
@@ -978,14 +981,39 @@ function checkSortAndFilter(e: SortChangedEvent) {
     .filter((sort) => sort)
   const filter = makeFilterModelList(gridFilterModel)
   if (sort.length || filter.length) {
-    isCreateNodeEnabled.value = true
     sortModel.value = sort as SortModel[]
     filterModel.value = filter
   } else {
-    isCreateNodeEnabled.value = false
     sortModel.value = []
     filterModel.value = []
   }
+  enableCreateNodeButton()
+}
+
+const onColumnStateChange = (e: ColumnVisibleEvent) => {
+  const colState = e.api.getColumnState()
+  hiddenColumns.value = colState.filter((col) => col.hide).map((col) => col.colId)
+  const gridColOrder = colState
+    .filter((col) => col.colId != INDEX_FIELD_NAME)
+    .map((col) => col.colId)
+  const defaultColOrder =
+    typeof props.data === 'object' && 'header' in props.data && props.data.header ?
+      props.data.header
+    : []
+  if (gridColOrder.every((val, index) => val === defaultColOrder[index])) {
+    vizColumnOrder.value = []
+  } else {
+    vizColumnOrder.value = gridColOrder
+  }
+  enableCreateNodeButton()
+}
+
+const enableCreateNodeButton = () => {
+  isCreateNodeButtonEnabled.value =
+    sortModel.value.length > 0 ||
+    filterModel.value.length > 0 ||
+    hiddenColumns.value.length > 0 ||
+    vizColumnOrder.value.length > 0
```

Comment:
This should be a `computed`--compared to an imperative update function, it prevents bugs that otherwise easily occur when the update function isn't run everywhere it needs to be.